clusterloader2/pkg/measurement/common/profile.go

```diff
@@ -218,6 +221,17 @@ func (p *profileMeasurement) shouldExposeApiServerDebugEndpoint() bool {
 	return p.config.componentName == "kube-apiserver"
 }
 
+func shouldGetAPIServerByK8sClient(componentName string) bool {
+	// In some cases the kube-apiserver cannot be reached by curl.
+	// We add a config here as a walkaround.
+	getAPIServerByK8sClient, err := strconv.ParseBool(os.Getenv("GET_APISERVER_PPROF_BY_K8S_CLIENT"))
```

Comment:
Wouldn't be cleaner to have a bool-env flag for this? Some advantages:

1. Parsing would be done at the init time, e.g. if the env is set incorrectly you'll now it immediately, not in the middle of some long test
2. Clean documented "api" for this feature, not some env hidden in the middle of some file